app/models/labware_creators/cardinal_pools_plate.rb

```diff
@@ -0,0 +1,139 @@
+# 2. Get the passed wells from the parent (LCA PBMC) plate
+# 3. For the number of passed wells, get the number of pools from config
+# e.g. if there are 96 passed wells on the parent, the samples get split into 8 pools, with 12 samples per pool
+# 4. Group samples by supplier, to ensure samples with the same supplier are distrubuted across different pools
+# 5. Add the pool to a well in the new LCA PBMC Pools plate
+module LabwareCreators
+  # This class is used for creating randomicardinal pools into destination plate
+  class CardinalPoolsPlate < Base
+    include SupportParent::PlateOnly
+    # include LabwareCreators::CustomPage
+
+    def filters=(filter_parameters)
+      well_filter.assign_attributes(filter_parameters)
+    end
+
+    # This should only be called from passed_parent_wells
+    # As we want to only transfer passed wells to the LCA PBMC plate
+    def labware_wells
+      parent.wells
+    end
+
+    # returns: a list of passed samples
+    def passed_parent_wells
+      labware_wells.select { |well| well.state == 'passed' }
+    end
+
+    # returns: the number of pools required for a given passed samples count
+    # this config is appended in the Cardinal initialiser
+    # e.g. 95,12,12,12,12,12,12,12,11 ==> 8
+    # e.g. 53,11,11,11,10,10,,, ==> 5
+    def number_of_pools
+      Rails.application.config.cardinal_pooling_config[passed_parent_wells.count]
+    end
+
+    def well_filter
+      @well_filter ||= WellFilter.new(creator: self)
+    end
+
+    # Send the transfer request to SS
+    def transfer_material_from_parent!(dest_uuid)
+      dest_plate = Sequencescape::Api::V2::Plate.find_by(uuid: dest_uuid)
+      api.transfer_request_collection.create!(
+        user: user_uuid,
+        transfer_requests: transfer_request_attributes(dest_plate)
+      )
+      true
+    end
+
+    # returns: a list of objects, mapping source well to destination well
+    # e.g [{'source_asset': 'auuid', 'target_asset': 'anotheruuid'}]
+    def transfer_request_attributes(dest_plate)
+      passed_parent_wells.map do |source_well, additional_parameters|
+        # TODO: revert back to the below
+        request_hash(source_well, dest_plate, additional_parameters)
+
+        # Temp fix below is for testing until we have the tag clash on pooling issue resolved
+        # as only returning the first request_hash for now whilst developing
+        # return [request_hash(source_well, dest_plate, additional_parameters)]
+      end
+    end
+
+    def request_hash(source_well, dest_plate, _additional_parameters)
+      {
+        'source_asset' => source_well.uuid,
+        'target_asset' => dest_plate.wells.detect do |dest_well|
+          dest_well.location == transfer_hash[source_well.location][:dest_locn]
+        end&.uuid,
+        # TODO: add concentration/ cell count here?
+        # 'volume' => "12345" #transfer_hash[source_well.location]['volume'].to_s
+        'aliquot_attributes': { 'tag_depth' => tag_depth(source_well) }
```

Comment:
I wonder whether this object should be part of additional_parameters? Passing into this method `'aliquot_attributes': { 'tag_depth' => tag_depth(source_well) }` as `additional_parameters` ?

Reply:
We can also remove volume if not needed

Reply:
Also, leave for now, think its clearer like this and we aren't sure exactly how additional_parameters are handled